lib/Dialect/ESI/runtime/cpp/include/esi/Services.h

```diff
@@ -284,6 +284,7 @@ class FuncService : public Service {
     std::mutex callMutex;
     WriteChannelPort *arg;
     ReadChannelPort *result;
+    bool connected = false;
```

Comment:
Instead of storing a third connected status, you could query the underlying read/write ports to see if they're both connected. In general, I prefer to avoid redundant state.

Reply:
Totally agree, but when digging into the sources, i was a bit unsure about what the general approach to `BundlePort`s' overall connectedness-status is. I suspect this isn't the only case where a `BundlePort` needs to connect to some underlying set of ports, which probably warrants something more generic...